File: server/src/main/java/gg/mineral/server/network/packet/play/bidirectional/KeepAlivePacket.java
package gg.mineral.server.network.packet.play.bidirectional;

import gg.mineral.api.network.connection.Connection;
import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Accessors(fluent = true)
public final class KeepAlivePacket implements Packet.INCOMING, Packet.OUTGOING {
    private int keepAliveId;

    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(keepAliveId);
    }

    @Override
    public void received(Connection connection) {
    }

    @Override
    public void deserialize(ByteBuf is) {
        keepAliveId = is.readInt();
    }

    @Override
    public byte getId() {
        return 0x00;
    }

}

File: server/src/main/java/gg/mineral/server/network/packet/play/bidirectional/KeepAlivePacket.kt
package gg.mineral.server.network.packet.play.bidirectional

import gg.mineral.api.network.connection.Connection
import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor
import lombok.experimental.Accessors

@NoArgsConstructor
@AllArgsConstructor
@Data
@Accessors(fluent = true)
class KeepAlivePacket : Packet.ASYNC_INCOMING, Packet.OUTGOING {
    private var keepAliveId = 0

    override fun serialize(os: ByteBuf) {
        os.writeInt(keepAliveId)
    }

    override fun received(connection: Connection) {
    }

    override fun deserialize(`is`: ByteBuf) {
        keepAliveId = `is`.readInt()
    }

    override fun getId(): Byte {
        return 0x00
    }

    override fun receivedAsync(connection: Connection) {
    }
}
